nexus/src/app/disk.rs

```diff
@@ -510,8 +510,11 @@ impl super::Nexus {
             .project_delete_snapshot(opctx, &authz_snapshot, &db_snapshot)
             .await?;
 
-        // Kick off volume deletion saga
+        // Kick off volume deletion saga(s)
         self.volume_delete(db_snapshot.volume_id).await?;
+        if let Some(volume_id) = db_snapshot.destination_volume_id {
```

Comment:
It seems like a couple of these steps maybe need to happen transactionally?  If we crash either after L511 or L514, it seems like we'd wind up not having cleaned stuff up.

I can understand punting on this but it seems like we should track this somewhere (even if just a TODO).

Reply:
This is a problem, yeah. `Nexus::project_delete_snapshot` is directly called from an HTTP endpoint, so it's not a case where a saga node would be replayed during a crash. I'm not sure how to solve this, and I feel like this may be a more general problem. I'll give it some thought.

Reply:
This should be addressed by https://github.com/oxidecomputer/omicron/pull/2090